src/app/core/entity-details/related-entities/related-entities.component.ts

```diff
@@ -36,54 +45,75 @@ export class RelatedEntitiesComponent<E extends Entity> implements OnInit {
   @Input()
   public set columns(value: ColumnConfig[]) {
     this._columns = value;
+    this.updateColumnsToDisplayForScreenSize();
   }
   public get columns(): ColumnConfig[] {
     return this._columns;
   }
   protected _columns: ColumnConfig[];
 
+  columnsToDisplay: string[];
+
   @Input() filter?: DataFilter<E>;
 
   @Input() showInactive: boolean;
 
-  data: E[] = [];
-  isLoading = false;
+  data: E[];
   private isArray = false;
   protected entityCtr: EntityConstructor<E>;
 
   constructor(
     protected entityMapper: EntityMapperService,
-    private entities: EntityRegistry,
-  ) {}
+    private entityRegistry: EntityRegistry,
+    private screenWidthObserver: ScreenWidthObserver,
+  ) {
+    this.screenWidthObserver
+      .shared()
+      .pipe(untilDestroyed(this))
+      .subscribe(() => this.updateColumnsToDisplayForScreenSize());
+  }
 
   async ngOnInit() {
     await this.initData();
+    this.listenToEntityUpdates();
   }
 
   protected async initData() {
-    this.isLoading = true;
-
-    this.entityCtr = this.entities.get(this.entityType) as EntityConstructor<E>;
+    this.entityCtr = this.entityRegistry.get(
+      this.entityType,
+    ) as EntityConstructor<E>;
     this.isArray = isArrayProperty(this.entityCtr, this.property);
 
-    this.data = (await this.entityMapper.loadType<E>(this.entityType)).filter(
-      (e) =>
-        this.isArray
-          ? e[this.property].includes(this.entity.getId())
-          : e[this.property] === this.entity.getId(),
-    );
     this.filter = {
       ...this.filter,
       [this.property]: this.isArray
         ? { $elemMatch: { $eq: this.entity.getId() } }
         : this.entity.getId(),
     };
 
+    this.data = (await this.entityMapper.loadType<E>(this.entityType)).filter(
+      (e) =>
+        this.isArray
+          ? e[this.property].includes(this.entity.getId())
+          : e[this.property] === this.entity.getId(),
+    );
+
     if (this.showInactive === undefined) {
       this.showInactive = this.entity.anonymized;
     }
+  }
 
-    this.isLoading = false;
+  private updateSubscription: Subscription;
+
+  protected listenToEntityUpdates() {
+    if (!this.updateSubscription && this.entityCtr) {
```

Comment:
Do we actually need this check if the function is only once called in `ngOnInit`?

Reply:
no, in most of these we don't. Thanks!